### ISSUE TITLE
Issue with auto-completion when dialyzer is disabled for an umbrella app

### ISSUE THREAD

**Issue by akash-akya** [elixir-lsp/elixir-ls#209]
### Environment
```
Erlang/OTP 21 [erts-10.3] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]
Elixir 1.7.3 (compiled with Erlang/OTP 21)
```
Language server version: v0.3.3

MacOS Mojave, Emacs with eglot as LS client

### Issue
Elixir LS fails to return autocompletion results for umbrella application when dialyzer is disabled

### Reproducing the issue
* clone https://github.com/akash-akya/elixir-ls-test
* open project with **dialyzer disabled**, with emacs and `eglot`, dialyzer is disabled by default using `.dir-locals.el`. 
* let elixir-ls build and generate .beam files
* open `apps/foo/lib/foo.ex` and type `Core.Bar` this should trigger auto-completion and it should show `Core.BarServer` as suggestion
* close the project and open it again
* try the same auto-completion as before, this time it won't show the suggestion

My best guess is, it is something to do with not triggering complete project compilation when we reopen the project.

**Comment by lukaszsamson** [comment]
I think it's related to the quirky way elixir-ls loads user modules. See https://github.com/elixir-lsp/elixir-ls/blob/master/guides/initialization.md#analyzing-the-project for details.

TLDR;
When dialyzer is not enabled Modules are not properly loaded and cannot be analysed hence no completions.

**Comment by akash-akya** [comment]
I see, thanks for the quick response.
If I understand it correctly, loading is done by dialyzer but the suggestion is provided ElixirSense depending on which modules are loaded? 

So "suggestions"  internally depend on the dialyzer. Can we use elixir_sense server and use `reload` to reload modules? (https://github.com/elixir-lsp/elixir_sense/blob/master/lib/elixir_sense/server/context_loader.ex#L25)?

**Comment by akash-akya** [comment]
I see, thanks for the quick response.

We have a big umbrella project, so I have to turn off the dialyzer. 
For now, I'm using [this ugly hack](https://github.com/akash-akya/elixir-ls/commit/2f3470beb6322bf989527eb8add3692470b5e668) and so far it seems to work.

**Comment by axelson** [comment]
@akash-akya thanks for showing the patch/hack you're using, that is helpful. Could you try replacing most of it by `Keyword.fetch!(Mix.Project.config(), :app) |> Application.load()`

**Comment by akash-akya** [comment]
Hi @axelson, I thought of something similar, but this does not works, as mix file of the project is not loaded yet. Basically `:app` will be nil.

I tried using `Mix.Project.in_project` to fix this. something like 
```elixir
  defp load_modules(state) do
    guessed_project_name = Path.basename(state.project_dir)

    Mix.Project.in_project(String.to_atom(guessed_project_name), state.project_dir, fn _project ->
      if Mix.Project.umbrella?() do
        Mix.Project.apps_paths()
        |> Map.keys()
        |> Enum.each(&Application.load/1)
      else
        Keyword.fetch!(Mix.Project.config(), :app)
        |> Application.load()
      end
    end)

    %{state | modules_loaded?: true}
  end
```
This does load the project, but now we get a compilation error when the compiler tries to build, 
```
== Compilation error in file mix.exs ==
** (CompileError) mix.exs:1: cannot define module RheaUmbrella.Mixfile because it is currently being defined in /Users/akash/repo/rhea/mix.exs:1
    (stdlib) erl_eval.erl:680: :erl_eval.do_apply/6
```

Moreover, this does not load modules from`.elixir-ls` build path. The default build path is often outdated.

**Comment by axelson** [comment]
Hmmm, okay. It would be great to fix this, but we would want to find a less hacky way if possible :)
